src/Cli/Action/PushRancherAction.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+/*
+ * This file is part of the Acme PHP project.
+ *
+ * (c) Titouan Galopin <galopintitouan@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace AcmePhp\Cli\Action;
+
+use AcmePhp\Cli\Serializer\PemEncoder;
+use AcmePhp\Ssl\CertificateResponse;
+use GuzzleHttp\Client;
+use Symfony\Component\Serializer\SerializerInterface;
+
+/**
+ * Action to upload SSL certificates to Rancher using its API.
+ *
+ * @see http://docs.rancher.com/rancher/v1.2/en/api/api-resources/certificate/
+ *
+ * @author Titouan Galopin <galopintitouan@gmail.com>
+ */
+class PushRancherAction implements ActionInterface
+{
+    /**
+     * @var Client
+     */
+    private $httpClient;
+
+    /**
+     * @var SerializerInterface
+     */
+    private $serializer;
+
+    /**
+     * @param Client              $httpClient
+     * @param SerializerInterface $serializer
+     */
+    public function __construct(Client $httpClient, SerializerInterface $serializer)
+    {
+        $this->httpClient = $httpClient;
+        $this->serializer = $serializer;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getName()
+    {
+        return 'push_rancher';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handle($config, CertificateResponse $response)
+    {
+        $payload = $this->createRancherPayloadFromResponse($response);
+
+        $commonName = $response->getCertificateRequest()->getDistinguishedName()->getCommonName();
+        $currentCertificates = $this->getRancherCertificates($config);
+
+        $updated = false;
+
+        foreach ($currentCertificates as $certificate) {
+            if ($certificate['name'] === $commonName) {
+                $updated = true;
+                $this->updateRancherCertificate($config, $certificate['id'], $payload);
+            }
+        }
+
+        if (!$updated) {
+            $this->createRancherCertificate($config, $payload);
+        }
+    }
+
+    private function createRancherPayloadFromResponse(CertificateResponse $response)
+    {
+        $certificate = $response->getCertificate();
+        $privateKey = $response->getCertificateRequest()->getKeyPair()->getPrivateKey();
+
+        // Issuer chain
+        $issuerChain = [];
+        $issuerCertificate = $certificate->getIssuerCertificate();
+
+        while (null !== $issuerCertificate) {
```

Comment:
We often use this code, what's about adding a method `Certificate::getChain` which return the array of certificates.